lib/IRGen/GenConstant.cpp

```diff
@@ -56,6 +56,24 @@ llvm::Constant *irgen::emitConstantInt(IRGenModule &IGM,
   return llvm::ConstantInt::get(IGM.getLLVMContext(), value);
 }
 
+llvm::Constant *irgen::emitConstantZero(IRGenModule &IGM, BuiltinInst *BI) {
+  assert(IGM.getSILModule().getBuiltinInfo(BI->getName()).ID ==
+         BuiltinValueKind::ZeroInitializer);
+  BuiltinIntegerWidth width =
+      BI->getType().castTo<AnyBuiltinIntegerType>()->getWidth();
+
+  APInt value(width.getLeastWidth(), 0);
+
+  // Handle arbitrary-precision integers.
+  if (width.isArbitraryWidth()) {
```

Comment:
Can you add a test for this case?

Reply:
Sure, is there a good arbitrary width constant test I can copy from?